src/lib/plugins/live-rows/live-rows.component.ts

```diff
@@ -0,0 +1,102 @@
+import { Component, OnInit, Input, ChangeDetectionStrategy } from '@angular/core';
+import { PluginService } from '../plugin.service';
+import { Fastdom } from '../../../core/services/fastdom';
+import { GRID_PREFIX } from 'ng2-qgrid/core/definition';
+import { jobLine } from 'ng2-qgrid/core/services/job.line';
+import { AppError } from '../../../../out-tsc/core/infrastructure/error';
+
+@Component({
+	selector: 'q-grid-live-rows',
+	template: '<ng-content></ng-content>',
+	providers: [PluginService],
+	changeDetection: ChangeDetectionStrategy.OnPush
+})
+export class LiveRowsComponent implements OnInit {
+	private currentRows: any[];
+	private previousRows: any[];
+	private animations = [];
+
+	@Input('duration') duration: number;
+
+	constructor(private plugin: PluginService) { }
+
+	ngOnInit() {
+		const { model } = this.plugin;
+		model.animation({
+			apply: (memo, context, complete) => {
+				this.previousRows = this.currentRows;
+				this.currentRows = memo.rows;
+
+				if (!this.previousRows || !this.currentRows) {
+					complete(memo);
+					return;
+				}
+
+				for (let i = 0; i < this.previousRows.length; i++) {
```

Comment:
for (let i = 0; length = ....; i < length

it's a good practice to do like that,